### PR TITLE
Audit improvements

### PR DESCRIPTION
**Pull request by Fang-**
Granular and standalone commits for your reviewing pleasure.

As per @iamwillkim's request:
2. Extraneous functions: These points make sense and we should do this
(we don’t need to break out short one-liners)
3. Documentation issues: all points valid, address all
5. Test coverage: Jared added a few more tests, so we have better
coverage, but we need two more tests to hit 100% branch coverage (one
for childSeedFromSeed() and one for get()).

This also makes management nodes per-ship rather than one per wallet/owner. That affects networking seeds, I'm not sure if I did the best possible job of capturing management seeds for use there. Please review with a strict eye on functional style.

### CODE REVIEW

File: src/index.js [hunk 1]
```diff
@@ -172,21 +145,21 @@ const childSeedFromSeed = async config => {
 
   const childSeed = await hash(seed, salt, defaultTo(password, ''));
 
-  return childSeed.slice(0, seed.length || seed.byteLength);
+  return childSeed.slice(0, seed.byteLength || seed.length);
```
Comment (Fang-) [inline src/index.js]: The fact that both of these get hit now (evidenced in test coverage) is weird. Buffers should always have a `byteLength` property as far as I'm aware, so the docstrings for this function may not be correct.

As far as `hash()` is concerned passing a string is fine, but may or may not have different output?

**Comment by Fang-** [comment]
There may be argon2-wasm interface changes incoming here: urbit/argon2-wasm#3. I will update this PR once those land.

File: src/index.js (hunk 1 above)
Comment (Fang-) [reply src/index.js]: The docstrings say this function accepts a `Buffer`. Calling this function with a `string` instead works. This seems confusing, and potentially dangerous if any function in the chain interprets strings and Buffers as different data.

File: src/index.js [hunk 2]
```diff
@@ -23,33 +23,6 @@ const splitAt = (index, str) => [str.slice(0, index), str.slice(index)];
 
 
 
-/**
- * Wraps Buffer.from(). Converts an array into a buffer.
- * @param  {array} arr
- * @return {buffer}
- */
-const bufferFrom = arr => Buffer.from(arr);
```
[the remark this answers is not in the document]
Comment (jtobin) [reply src/index.js]: `bufferFrom`, we hardly knew ye.

File: src/index.js [hunk 3]
```diff
@@ -374,19 +374,18 @@ const fullWalletFromTicket = async config => {
...
 
-  let manageSeeds = {};
   const manageNodes = await Promise.all(ships.map(async ship => {
-    let res = await childNodeFromSeed({
-      seed: ownerSeed,
-      type: 'manage',
-      revision: _revisions.manage,
-      ship: ship,
-      password: password,
+     return await childNodeFromSeed({
```
Comment (Fang-) [inline src/index.js]: Pretty sure we can then do away with the `async` and `{ return await` and `}` to get us back to that lean, mean, functional aesthetic.

Great change @jtobin!

Comment (jtobin) [reply src/index.js]: @Fang- right you are.  Slash them LoC!

Comment (Fang-) [reply src/index.js]: For the record, I am generally in favor of putting in the optional `{}` and `return`, for the same reason I'm in favor of optional `()`. But right here and now, that would be inconsistent with the rest of the codebase.

Comment (jtobin) [reply src/index.js]: I may opine on this topic in a future issue of the Digest.  In general, I think matching the surrounding style of the code is the most important thing.  People can argue, very reasonably on all sides, that something should or should not be this way or that, syntactically or stylewise or what have you.

(Except tabs vs spaces, of course -- people who advocate tabs are beyond the reach of reason.)

But I think the author/maintainer of a codebase ought set the style, whatever that is, and that contributors ought match it.  An author/maintainer who enforces no particular style, or who accepts contributions of any style willy-nilly, is probably doing everyone a disservice.

**Pull request merged**